### PR TITLE
Add login

### PR DESCRIPTION
**Pull request by Ostapenco**
@dumitru-stratulat

### CODE REVIEW

File: src/pages/login/index.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        password: password,
+      });
+      return 200;
+    } catch (err) {
+      console.error(err);
+    }
+  };
+
+  const onFinish = async () => {
+    const status = await auth(email, password);
```
Comment (dumitru-stratulat) [inline src/pages/login/index.tsx]: catch input value by name prop in <Form.Item> tag in function onFinish by first arg. ex: onFinish = async (value)=>{auth(value)}

File: src/pages/signup/index.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        password: password,
+      });
+      return 201;
+    } catch (err) {
+      console.error(err);
+    }
+  };
+
+  const onFinish = async () => {
+    const status = await auth(email, username, password);
```
Comment (dumitru-stratulat) [inline src/pages/signup/index.tsx]: catch input value by name prop in <Form.Item> tag in function onFinish by first arg. ex: onFinish = async (value)=>{auth(value)}

File: src/pages/login/index.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
+import React, { useState } from "react";
+import axios from "axios";
+import Router from "next/router";
+import { Form, Input, Button, Checkbox } from "antd";
+import { UserOutlined, LockOutlined } from "@ant-design/icons";
+import Link from "next/Link";
+
```
Comment (dumitru-stratulat) [inline src/pages/login/index.tsx:7]: remove all empty rows between imports

File: src/pages/login/index.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        });
+
+      return 200;
+    } catch (err) {
+      console.error(err);
+    }
+  };
+
+  const onFinish = async (value: Value) => {
+    console.log("onFinish -> value", value);
```
Comment (AleVul) [inline src/pages/login/index.tsx]: please remove this log

File: src/pages/login/index.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+import style from "./login.module.css";
+
+interface Value {
+  email: string;
+  password: string;
+}
+
+const Login = () => {
+  const auth = async (value: Value) => {
```
Comment (AleVul) [inline src/pages/login/index.tsx:18]: you dont need this function to be in the `Login` component, suggest to move it outside

File: src/pages/login/index.tsx [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    try {
+      await axios
+        .post("https://reactive.loca.lt/login/", {
+          email: value.email,
+          password: value.password,
+        })
+        .then(function (res) {
```
Comment (AleVul) [inline src/pages/login/index.tsx]: you don't need `then` if you are using `async/await` syntax, just continue the logic as if a synchronous function is called

File: src/pages/signup/index.tsx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+  const auth = async (value: Value) => {
+    try {
+      await axios.post("https://reactive.loca.lt/signup/", {
+        email: value.email,
+        username: value.username,
+        password: value.password,
+      });
+      return 201;
+    } catch (err) {
+      console.error(err);
```
Comment (AleVul) [inline src/pages/signup/index.tsx]: log to console are not proper error handling, in such cases i usually use `status` flags, for example:

- pending - when a request has been made and we are waiting for a response
- success - when we receive the expected success response from server
- error - when we receive an error as a response, usually anything with status code 400 +

an depending on the status, we can render different things

File: src/pages/login/index.tsx [hunk 8]
```diff
@@ -13,27 +13,25 @@ interface Value {
 }
 
 const Login = () => {
+  const [error, setError] = useState();
+  const [loading, setLoading] = useState(false);
```
Comment (AleVul) [inline src/pages/login/index.tsx]: Suggest to join this 2 in one state

for example:
`type Status = "pending" | "success" | "error";

...

const [status, setStatus] = useState<Status>("pending");`

Comment (AleVul) [reply src/pages/login/index.tsx]: If you want to keep error message, you could use an [union](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions)

**Pull request merged**